### DISCUSSION TITLE
Access token failure

### DISCUSSION THREAD

**Discussion by ktoonsez** [timdorr/tesla-api#380] [Q&A]
As of this morning I keep throwing this error:

Request not succesful: Response{protocol=h2, code=400, message=, url=https://auth.tesla.com/oauth2/v3/authorize/mfa/verify}.

Anyone else?

**Comment by hemang249** [comment]
Nope, scripts on my end are functional. Can you share the exact request and the headers and the bodyyou are sending? 
A 400 status generally means you request is missing something

**Reply by ktoonsez** [reply]
Been using the same code for months so not sure why it would stop working. I will capture the headers and body in debug and post it up?  So nothings changed in the required headers in last 30 days since my token just expired yesterday?

**Reply by hemang249** [reply]
Ok update: I checked the logs of my script from today afternoon and I too am getting a 400 status now. The exact error seems to be pointing towards the client_id being invalid. This could either be a bug on Tesla's end or a deliberate attempt to block of unauthorized use of the API.

```
<Response [400]>
{'error': 'unauthorized_client', 'error_description': 'Unauthorized client', 'error_uri': 'https://auth.tesla.com/error/reference/83ac7c2f-c99d-40ba-965a-56e474e2e785-1621438034202'}
```

**Reply by ktoonsez** [reply]
Ok, I figured it couldn't just be me only, that makes more since now. I did try changing my client ID to see if somehow I got banned and I got the same 400 last night.

**Reply by enginehead-tes** [reply]
The same issue with me. It happened after the token expired and I needed a new one. Also same problem with someone else who is using my app. *You don't have permission to access "http://auth.tesla.com/oauth2/v3/token" on this server.<P> Reference #18.4c0f0317.1621505359.3c408e5e* is the error.
It happens on step 3. The value for hidden field "cancel" is now empty I'm not sure if it used to be or not. Maybe some other "hidden" value required?

**Reply by SeBsZ** [reply]
Same issue here, not sure since how long but probably a few days. Does anyone have a solution?

**Reply by ktoonsez** [reply]
Nothing here, I have tried a few thing with no luck.

**Reply by hemang249** [reply]
I have tried various different client_ids, None have worked so far.  The other endpoints work fine with the same client_id. So I am not sure if this is a Bug on Tesla's end or they are doing it on purpose

**Reply by enginehead-tes** [reply]
If anyone needs a quick fix there's an IOS app called "AuthAppForTesla" which appears to be working. Token valid for over a month.

**Reply by ktoonsez** [reply]
Still no luck over here, anyone else?
